src/annotator/anchoring/types.js

```diff
@@ -0,0 +1,218 @@
+/**
+ * This module exports a set of classes for converting between DOM `Range`
+ * objects and different types of selectors. It is mostly a thin wrapper around a
+ * set of anchoring libraries. It serves two main purposes:
+ *
+ *  1. Providing a consistent interface across different types of anchors.
+ *  2. Insulating the rest of the code from API changes in the underlying anchoring
+ *     libraries.
+ */
+import {
+  fromRange as posFromRange,
+  toRange as posToRange,
+} from 'dom-anchor-text-position';
+import {
+  fromRange as quoteFromRange,
+  toRange as quoteToRange,
+  toTextPosition,
+} from 'dom-anchor-text-quote';
+
+import { SerializedRange, sniff } from './range';
+
+/**
+ * @typedef {import("./range").BrowserRange} BrowserRange}
+ * @typedef {import("./range").NormalizedRange} NormalizedRange}
+ * @typedef {BrowserRange|NormalizedRange|SerializedRange} AnyRangeType
+ *
+ * @typedef {import('../../types/api').RangeSelector} RangeSelector
+ * @typedef {import('../../types/api').TextPositionSelector} TextPositionSelector
+ * @typedef {import('../../types/api').TextQuoteSelector} TextQuoteSelector
+ *
+ * @typedef TextContentNode
+ * @prop {string} textContent
+ */
+
+/**
+ * @typedef SelectorOptions
```

Comment:
The different options here refer to different selector types. I would suggest to remove this and document the supported properties for each `options` usage as appropriate. In other words, something like this:

```js
class RangeAnchor {
  /**
   * @param {Object} [options]
   * @param {string} [options.ignoreSelector]
   */
  toSelector(options = {}) {

  }
}
```

Reply:
makes sense